web-common/src/features/entity-management/watch-files-client.ts

```diff
@@ -0,0 +1,62 @@
+import { WatchRequestClient } from "@rilldata/web-common/runtime-client/watch-request-client";
+import {
+  getRuntimeServiceGetFileQueryKey,
+  getRuntimeServiceListFilesQueryKey,
+  V1WatchFilesResponse,
+} from "@rilldata/web-common/runtime-client";
+import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import type { QueryClient } from "@tanstack/svelte-query";
+import { get } from "svelte/store";
+
+export function watchFilesClient(queryClient: QueryClient) {
+  return new WatchRequestClient<V1WatchFilesResponse>(
+    (runtime) =>
+      `${runtime.host}/v1/instances/${runtime.instanceId}/files/watch`,
+    (res) => invalidateFile(queryClient, res),
+    () => invalidateAllFiles(queryClient)
+  ).start();
+}
+
+function invalidateFile(queryClient: QueryClient, res: V1WatchFilesResponse) {
+  // Watch file returns events for all files under the project. Ignore everything except .sql, .yaml & .yml
+  if (
+    !res.path.endsWith(".sql") &&
+    !res.path.endsWith(".yaml") &&
+    !res.path.endsWith(".yml")
+  )
+    return;
```

Comment:
I think we should remove this condition. This work will enable us to show the whole project directory in the sidebar, including `.md`, `.env`, and `.py` files.

Until we get there, I think the filtering for specific file types should be logic in the sidebar component, not in this client.

Reply:
I disagree. Lets add to this filter once we integrate the new editor properly. Right now we get 3+ events of updated to `.db.wal` for every file write.

Reply:
Sure, that works